airflow/www/views.py

```diff
@@ -963,6 +963,42 @@ def run(self):
             "it should start any moment now.".format(ti))
         return redirect(origin)
 
+    @expose('/trigger')
```

Comment:
This function is very similar to trigger_dag in the CLI. This is why we need an API, so common functionality can be used between the CLI and Web. Do you have an idea of how to reuse this code as per DRY? The main differences are flash vs logging calls.